src/test/regress/input/multi_view.source

```diff
@@ -0,0 +1,164 @@
+	WHERE (o_orderkey = 231);
+
+-- select distinct on router joins of views also works
+SELECT distinct(o_orderkey) FROM priority_orders join air_shipped_lineitems
+	ON (o_orderkey = l_orderkey)
+	WHERE (o_orderkey = 231);
+
+-- left join support depends on flattening of the query
+-- following query fails since the inner part is kept as subquery
+SELECT * FROM priority_orders left join air_shipped_lineitems ON (o_orderkey = l_orderkey);
+
+-- however, this works
+SELECT count(*) FROM priority_orders left join lineitem_hash ON (o_orderkey = l_orderkey) WHERE l_shipmode ='AIR';
+
+-- view at the inner side of is not supported
+SELECT count(*) FROM priority_orders right join lineitem_hash ON (o_orderkey = l_orderkey) WHERE l_shipmode ='AIR';
+
+-- but view at the outer side is. This is essentially the same as a left join with arguments reversed.
+SELECT count(*) FROM lineitem_hash right join priority_orders ON (o_orderkey = l_orderkey) WHERE l_shipmode ='AIR';
+
+
+-- left join on router query is supported
+SELECT o_orderkey, l_linenumber FROM priority_orders left join air_shipped_lineitems ON (o_orderkey = l_orderkey)
+	WHERE o_orderkey = 2;
+
+-- repartition query on view join
+-- it passes planning, fails at execution stage
+SELECT * FROM priority_orders JOIN air_shipped_lineitems ON (o_custkey = l_suppkey);
+
+SET citus.task_executor_type to "task-tracker";
+SELECT count(*) FROM priority_orders JOIN air_shipped_lineitems ON (o_custkey = l_suppkey);
+SET citus.task_executor_type to DEFAULT;
+
+-- insert into... select works with views
+CREATE TABLE temp_lineitem(LIKE lineitem_hash);
+SELECT create_distributed_table('temp_lineitem', 'l_orderkey');
+INSERT INTO temp_lineitem SELECT * FROM air_shipped_lineitems;
+SELECT count(*) FROM temp_lineitem;
+-- following is a where false query, should not be inserting anything
+INSERT INTO temp_lineitem SELECT * FROM air_shipped_lineitems WHERE l_shipmode = 'MAIL';
+SELECT count(*) FROM temp_lineitem;
+
+-- modifying views is disallowed
+INSERT INTO air_shipped_lineitems SELECT * from temp_lineitem;
+
+SET citus.task_executor_type to "task-tracker";
+
+-- single view repartition subqueries are not supported
+SELECT l_suppkey, count(*) FROM
+	(SELECT l_suppkey, l_shipdate, count(*)
+		FROM air_shipped_lineitems GROUP BY l_suppkey, l_shipdate) supps
+	GROUP BY l_suppkey ORDER BY 2 DESC, 1 LIMIT 5;
+
+-- logically same query without a view works fine
+SELECT l_suppkey, count(*) FROM
+	(SELECT l_suppkey, l_shipdate, count(*)
+		FROM lineitem_hash WHERE l_shipmode = 'AIR' GROUP BY l_suppkey, l_shipdate) supps
+	GROUP BY l_suppkey ORDER BY 2 DESC, 1 LIMIT 5;
+
+-- when a view is replaced by actual query it still fails
+SELECT l_suppkey, count(*) FROM
+	(SELECT l_suppkey, l_shipdate, count(*)
+		FROM (SELECT * FROM lineitem_hash WHERE l_shipmode = 'AIR') asi
+		GROUP BY l_suppkey, l_shipdate) supps
+	GROUP BY l_suppkey ORDER BY 2 DESC, 1 LIMIT 5;
+
+SET citus.task_executor_type to DEFAULT;
```

Comment:
Can we please add a test that includes a GROUP BY? I suspect those are actually the most frequent ones in practice, where the view "returns" some highly aggregated data, and you just perform some selection ontop of that.